services/src/pro/layers/postgres_layer_db.rs

```diff
@@ -480,10 +480,15 @@ where
             )
             .await?;
 
-        let id = provider.id();
+        let id = DataProviderDefinition::<Self>::id(&provider);
         conn.execute(
             &stmt,
-            &[&id, &provider.type_name(), &provider.name(), &provider],
+            &[
+                &id,
+                &DataProviderDefinition::<Self>::type_name(&provider),
+                &DataProviderDefinition::<Self>::name(&provider),
+                &provider,
+            ],
```

Comment:
Why can't we easily access the property functions anymore?

Reply:
If I understand correctly, `TypedDataProviderDefinition` now implements `DataProviderDefinition`.
`DataProviderDefinition` seems to already have the methods `type_name`, `name`, etc.
However, then you implement these methods again in `TypedDataProviderDefinition` and the compiler now does not know which method to call.

Couldn't you just remove the impls in `TypedDataProviderDefinition`?

Reply:
the problem is that there are now two `DataProviderDefinition` traits, and you have to tell the compiler which one you want. If I remove the impls from `TypedDataProviderDefinition` then I have to do the matching in both free and pro DB. And then the compiler still doesn't know the type

![image](https://github.com/geo-engine/geoengine/assets/10283892/974de5da-b1d1-42c5-8777-b625381c3ee3)